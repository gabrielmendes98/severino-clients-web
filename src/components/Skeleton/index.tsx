import MuiSkeleton, { SkeletonProps } from '@mui/material/Skeleton';
import { ReactNode } from 'react';
import Grid from '@mui/material/Grid';
import useAppTheme from 'common/hooks/useTheme';

interface Props {
  count?: number;
  SkeletonItem?: SkeletonProps;
  children?: ReactNode;
  ready: boolean;
  spacing?: number;
}

const Skeleton = ({
  count = 1,
  SkeletonItem,
  children,
  ready,
  spacing,
}: Props) => {
  const theme = useAppTheme();

  return (
    <>
      {ready ? (
        children
      ) : (
        <Grid container item spacing={spacing}>
          {Array.from(Array(count)).map((_, index) => (
            <Grid item key={index}>
              <MuiSkeleton
                {...SkeletonItem}
                width={theme.spacing(Number(SkeletonItem?.width))}
                height={theme.spacing(Number(SkeletonItem?.height))}
              />
            </Grid>
          ))}
        </Grid>
      )}
    </>
  );
};

export default Skeleton;
