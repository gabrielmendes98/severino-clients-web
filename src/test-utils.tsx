import React, { FC, ReactElement, ReactNode } from 'react';
import { render, RenderOptions } from '@testing-library/react';
import { Provider } from 'react-redux';
import { PreloadedState } from '@reduxjs/toolkit';
import ThemeProvider from './common/providers/Theme';
import { AppStore, RootState, setupStore } from './common/store/store';

interface ExtendedRenderOptions extends Omit<RenderOptions, 'queries'> {
  preloadedState?: PreloadedState<RootState>;
  store?: AppStore;
}

const customRender = (
  ui: ReactElement,
  {
    preloadedState = {},
    // Automatically create a store instance if no store was passed in
    store = setupStore(preloadedState),
    ...renderOptions
  }: ExtendedRenderOptions = {},
) => {
  const Wrapper: FC<{ children: ReactNode }> = ({ children }) => (
    <Provider store={store}>
      <ThemeProvider>{children}</ThemeProvider>
    </Provider>
  );

  return { ...render(ui, { wrapper: Wrapper, ...renderOptions }) };
};

/* eslint-disable */
export * from '@testing-library/react';
export { default as userEvent } from '@testing-library/user-event';
export { customRender as render };
/* eslint-enable */
