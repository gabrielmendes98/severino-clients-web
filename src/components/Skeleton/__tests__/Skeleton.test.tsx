import { render } from 'test-utils';
import Skeleton from '../index';

it('should match snapshot when ready', () => {
  const { container } = render(
    <Skeleton ready>
      <div>children</div>
    </Skeleton>,
  );
  expect(container).toMatchSnapshot();
});

it('should match snapshot when not ready', () => {
  const { container } = render(
    <Skeleton ready={false}>
      <div>children</div>
    </Skeleton>,
  );
  expect(container).toMatchSnapshot();
});
