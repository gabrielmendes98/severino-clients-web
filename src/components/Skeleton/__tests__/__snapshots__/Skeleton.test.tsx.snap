// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot when not ready 1`] = `
<div>
  <div
    class="Toastify"
  />
  <div
    class="MuiGrid-root MuiGrid-container MuiGrid-item css-1f064cs-MuiGrid-root"
  >
    <div
      class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
    >
      <span
        class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-pulse css-1n7gpw1-MuiSkeleton-root"
      />
    </div>
  </div>
</div>
`;

exports[`should match snapshot when ready 1`] = `
<div>
  <div
    class="Toastify"
  />
  <div>
    children
  </div>
</div>
`;
